test/index.test.js

```diff
@@ -0,0 +1,211 @@
+</builders>
+<publishers/>
+<buildWrappers/>
+</project>`;
+
+describe('index', () => {
+    let executor;
+    let Executor;
+    let fsMock;
+    let jenkinsMock;
+    let breakerMock;
+    let createOpts;
+    let buildOpts;
+
+    /**
+     * Stub for Breaker factory method
+     */
+    function BreakerFactory() {}
+
+    const config = {
+        buildId: '4b8d9b530d2e5e297b4f470d5b0a6e1310d29c5e',
+        jobId: '50dc14f719cdc2c9cb1fb0e49dd2acc4cf6189a0',
+        pipelineId: 'ccc49349d3cffbd12ea9e3d41521480b4aa5de5f',
+        container: 'node:4',
+        scmUrl: 'git@github.com:screwdriver-cd/data-model.git#master'
+    };
+
+    const jobName = config.buildId;
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+    });
+
+    beforeEach(() => {
+        fsMock = {
+            readFile: sinon.stub()
+        };
+
+        jenkinsMock = {
+            job: {
+                create: sinon.stub(),
+                build: sinon.stub()
+            }
+        };
+
+        breakerMock = {
+            runCommand: sinon.stub()
+        };
+
+        mockery.registerMock('fs', fsMock);
+    });
+
+    afterEach(() => {
+        mockery.deregisterAll();
+        mockery.resetCache();
+    });
+
+    after(() => {
+        mockery.disable();
+    });
+
+    it('extends base class', () => {
+        // eslint-disable-next-line global-require
+        Executor = require('../index');
+
+        executor = new Executor({
+            host: 'jenkins',
+            username: 'admin',
+            password: 'fakepassword'
+        });
+
+        // eslint-disable-next-line global-require
+        const BaseExecutor = require('../node_modules/screwdriver-executor-base/index');
+
+        assert.instanceOf(executor, BaseExecutor);
+    });
+
+    describe('start', () => {
+        beforeEach(() => {
+            BreakerFactory.prototype.runCommand = breakerMock.runCommand;
+            mockery.registerMock('circuit-fuses', BreakerFactory);
+
+            // eslint-disable-next-line global-require
+            Executor = require('../index');
+
+            executor = new Executor({
+                host: 'jenkins',
+                username: 'admin',
+                password: 'fakepassword'
+            });
+
+            createOpts = {
+                module: 'job',
+                action: 'create',
+                params: [jobName, TEST_XML]
+            };
+
+            buildOpts = {
+                module: 'job',
+                action: 'build',
+                params: [jobName]
+            };
+        });
+
+        it('return null when the job is successfully created', (done) => {
+            fsMock.readFile.yieldsAsync(null, TEST_XML);
+            breakerMock.runCommand.yieldsAsync(null, null);
+
+            executor.start(config, (err) => {
+                assert.isNull(err);
+                assert.calledOnce(fsMock.readFile);
```

Comment:
Unless this isn't really fully fleshed out, you should test that fs.readFile is called correctly.